moveit_commander/src/moveit_commander/move_group.py

```diff
@@ -597,14 +598,21 @@ def go(self, joints=None, wait=True):
             except TypeError:
                 self.set_joint_value_target(joints)
         if wait:
-            return self._g.move()
+            if planning_pipeline and planner_id:
+                return self._g.move(planning_pipeline, planner_id)
+            else:
+                return self._g.move()
         else:
-            return self._g.async_move()
+            if planning_pipeline and planner_id:
+                return self._g.async_move(planning_pipeline, planner_id)
+            else:
+                return self._g.async_move()
 
-    def plan(self, joints=None):
-        """Return a tuple of the motion planning results such as
+    def plan(self, joints=None, planning_pipeline=None, planner_id=None):
+        """Return a tuple of the motion planning results of the format:
         (success flag : boolean, trajectory message : RobotTrajectory,
-         planning time : float, error code : MoveitErrorCodes)"""
+         planning time : float, error code : MoveitErrorCodes)
+        planning_pipeline and planner_id specify the planner that is used for this plan."""
```

Comment:
Could you please also explain where the default values come from when these values are None?

Reply:
Done